lang/c++/impl/NodeImpl.cc

```diff
@@ -132,11 +132,19 @@ NodePrimitive::resolve(const Node &reader) const {
 
         case AVRO_BYTES:
 
-            return reader.type() == AVRO_STRING ? RESOLVE_MATCH : RESOLVE_NO_MATCH;
+            if (reader.type() == AVRO_STRING) {
+                return RESOLVE_MATCH;
+            }
+
+            break;
 
         case AVRO_STRING:
 
-            return reader.type() == AVRO_BYTES ? RESOLVE_MATCH : RESOLVE_NO_MATCH;
+            if (reader.type() == AVRO_BYTES) {
+                return RESOLVE_MATCH;
+            }
```

Comment:
nit: I don't know to much about AVRO, but do we need to make sure `RESOLVE_NO_MATCH` is returned somewhere? I'd like to believe the original devs were returning that value for a reason.

Reply:
That code isn't upstream, I added it here: https://github.com/redpanda-data/avro/pull/1

After this quick check, it goes into [furtherResolve](https://github.com/redpanda-data/avro/pull/119/files#diff-664aa9e29132f5bdd580b231ab8f159f8c5e6002496628248911c39b0e71e830R152).

All of that is then wrapped here: https://github.com/redpanda-data/redpanda/blob/v22.3.9/src/v/pandaproxy/schema_registry/avro.cc#L49-L51